src/Input/ExcludePathFilter.php

```diff
@@ -67,12 +84,25 @@ class ExcludePathFilter implements Filter
     public function __construct(array $patterns)
     {
         $quoted = array_map('preg_quote', $patterns);
-        $pattern = strtr(implode('|', $quoted), [
+        $patternString = strtr(implode('|', $quoted), [
             '\\*' => '.*',
             '\\\\' => '/',
         ]);
 
-        $this->pattern = '(^(' . $pattern . '))i';
+        if (empty($patterns) || $patternString === '') {
+            $this->pattern = '/^$/';
+
+            return;
+        }
+
+        if (strlen($patternString) > self::PATTERN_SPLIT_LIMIT) {
+            $this->isBulk = true;
+            foreach ($quoted as $pattern) {
+                $this->iterativePatterns[] = '(^(' . $pattern . '))i';
+            }
+        } else {
+            $this->pattern = '(^(' . $patternString . '))i';
+        }
```

Comment:
```suggestion
        if (strlen($patternString) <= self::PATTERN_SPLIT_LIMIT) {
                    $this->pattern = '(^(' . $patternString . '))i';
                    return;
        }
        
        $this->isBulk = true;
        foreach ($quoted as $pattern) {
            $this->iterativePatterns[] = '(^(' . $pattern . '))i';
        }

```

Prevent the else with an early return 😃

Reply:
ooh nice one, and thanks for the Suggest format - I'd almost cheekily click Commit Suggestion if not for the indentation haha! (and wanting to run the tests locally)

all done now.

Just a heads up, not sure if I need to do anything after approval, so please let me know if so.

Reply:
Once you get approval from two maintainers you won't have to do anything else.